addons/l10n_it_edi/models/account_move.py

```diff
@@ -0,0 +1,1212 @@
+    def _compute_show_reset_to_draft_button(self):
+        super()._compute_show_reset_to_draft_button()
+        for move in self.filtered(lambda m: m.l10n_it_edi_transaction):
+            move.show_reset_to_draft_button = False
+
+    def _get_edi_decoder(self, file_data, new=False):
+        # EXTENDS 'account'
+        if file_data['type'] == 'l10n_it_edi':
+            return self._l10n_it_edi_import
+        return super()._get_edi_decoder(file_data, new=new)
+
+    def _post(self, soft=True):
+        super()._post(soft)
+        for move in self:
+            move.l10n_it_edi_header = False
+
+    def _l10n_it_edi_get_line_values(self, reverse_charge_refund=False, is_downpayment=False, convert_to_euros=True):
+        """ Returns a list of dictionaries passed to the template for the invoice lines (DettaglioLinee)
+        """
+        invoice_lines = []
+        lines = self.invoice_line_ids.filtered(lambda l: l.display_type not in ('line_note', 'line_section'))
+        for num, line in enumerate(lines):
+            sign = -1 if line.move_id.is_inbound() else 1
+            price_subtotal = (line.balance * sign) if convert_to_euros else line.price_subtotal
+            # The price_subtotal should be inverted when the line is a reverse charge refund.
+            if reverse_charge_refund:
+                price_subtotal = -price_subtotal
+
+            # Unit price
+            price_unit = 0
+            if line.quantity and line.discount != 100.0:
+                price_unit = price_subtotal / ((1 - (line.discount or 0.0) / 100.0) * abs(line.quantity))
+            else:
+                price_unit = line.price_unit
+
+            description = line.name
+
+            # Down payment lines:
+            # If there was a down paid amount that has been deducted from this move,
+            # we need to put a reference to the down payment invoice in the DatiFattureCollegate tag
+            downpayment_moves = self.env['account.move']
+            if not is_downpayment and line.price_subtotal < 0:
+                downpayment_moves = line._get_downpayment_lines().mapped("move_id")
+                if downpayment_moves:
+                    downpayment_moves_description = ', '.join(m.name for m in downpayment_moves)
+                    sep = ', ' if description else ''
+                    description = f"{description}{sep}{downpayment_moves_description}"
+
+            invoice_lines.append({
+                'line': line,
+                'line_number': num + 1,
+                'description': description or 'NO NAME',
+                'unit_price': price_unit,
+                'subtotal_price': price_subtotal,
+                'vat_tax': line.tax_ids._l10n_it_filter_kind('vat'),
+                'downpayment_moves': downpayment_moves,
+                'discount_type': (
+                    'SC' if line.discount > 0
+                    else 'MG' if line.discount < 0
+                    else False)
```

Comment:
wrongly indented and not very readable
```suggestion
                'discount_type': (
                    'SC' if line.discount > 0
                    else 'MG' if line.discount < 0
                    else False)
```